spec/models/miq_report_result_spec.rb

```diff
@@ -1,14 +1,62 @@
 describe MiqReportResult do
-  context "persisting generated report results" do
-    before(:each) do
-      EvmSpecHelper.local_miq_server
+  before :each do
+    EvmSpecHelper.local_miq_server
+
+    @user1 = FactoryGirl.create(:user_with_group)
+  end
+
+  context "report result created by User 1 with current group 1" do
+    before :each do
+      @report_1 = FactoryGirl.create(:miq_report)
+      group_1 = FactoryGirl.create(:miq_group)
+      group_2 = FactoryGirl.create(:miq_group)
+      @user1.miq_groups << group_1
+      @report_result1 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report_1.id, :miq_group => group_1)
+      @report_result2 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report_1.id, :miq_group => group_1)
+      @report_result_nil_report_id = FactoryGirl.create(:miq_report_result)
+
+      @report_2 = FactoryGirl.create(:miq_report)
+      @report_result3 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report_2.id, :miq_group => group_2)
+      User.current_user = @user1
+    end
+
+    describe ".with_report" do
+      it "returns report all results without nil report_id" do
```

Comment:
@lpichler In a followup PR:
"returns report all results without nil report_id"
=> "returns all non-nil report_id results by default"

Or something like that...